internal/common/resource.go

```diff
@@ -93,8 +97,45 @@ func createOrUpdate(request *Request, resource controllerutil.Object, isClusterR
 
 func setOwner(request *Request, resource controllerutil.Object, isClusterRes bool) error {
 	if isClusterRes {
-		return libhandler.SetOwnerAnnotations(request.Instance, resource)
+		err := libhandler.SetOwnerAnnotations(request.Instance, resource)
+		if err != nil {
+			return err
+		}
+
+		// Removing ownerReferences for objects prior to creation
+		resource.SetOwnerReferences(nil)
+
+		// Removing ownerReferences for existing objects
+		existingRes := newEmptyResource(resource)
+		key, err := client.ObjectKeyFromObject(resource)
+		if err != nil {
+			return err
+		}
+
+		err = request.Client.Get(request.Context, key, existingRes)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+		} else {
+			if existingRes.GetOwnerReferences() != nil && len(existingRes.GetOwnerReferences()) > 0 {
+				patch := client.RawPatch(types.JSONPatchType,
+					[]byte(`[{"op": "replace","path": "/metadata/ownerReferences", "value": []}, {"op": "remove","path": "/metadata/ownerReferences"}]`))
+				err = request.Client.Patch(request.Context, existingRes, patch)
+
+				// If the OwnerReferences field does not exist for 'existingRes' that should be accepted as well,
+				// I couldn't find the typed error for this case (at least for unit-tests), so I went with
```

Comment:
There is no explicit typed error for this, but the jsonPatch wraps the error to this form here:
https://github.com/kubernetes/kubernetes/blob/0a839c6c3b9489716e216b4fc9dd2f9c17948c16/vendor/github.com/evanphx/json-patch/patch.go#L401
You could use that ErrMissing here, but I don't think if it should be a dependency for us as it's not a dep directly on K8s but a third_party library.